Dockerfile

```diff
@@ -0,0 +1,45 @@
+# Build in disposable container so run-time container is small
+FROM alpine:latest as build
+
+# Build from master by default but allow build time specification
+ARG ref=master
+ENV my_ref=$ref
+
+# Developers may wish to specify an alternate repository for source
+ARG repo=https://github.com/xmrig/xmrig.git
+ENV my_repo=$repo
+
+RUN set -ex && \
+        # testing required for hwloc
+	echo @testing http://nl.alpinelinux.org/alpine/edge/testing >> /etc/apk/repositories
```

Comment:
Did you get `hwloc-dev` to work coming from `testing`? I think it should come from `community`: https://alpine.pkgs.org/edge/alpine-community-x86_64/hwloc-dev-2.2.0-r0.apk.html 

```
Repository | Alpine Community x86_64 Official
-- | --
```